surefire-its/src/test/java/org/apache/maven/surefire/its/JUnitPlatformRerunFailingTestsIT.java

```diff
@@ -43,22 +42,22 @@ private SurefireLauncher unpack()
     @Test
     public void testRerunFailingErrorTestsWithOneRetry() throws Exception
     {
-        OutputValidator outputValidator = unpack().addGoal( GOAL ).setJUnitVersion( VERSION ).maven().addGoal(
+        OutputValidator outputValidator = unpack().setJUnitVersion( VERSION ).maven().addGoal(
                 "-Dsurefire.rerunFailingTestsCount=1" ).withFailure().executeTest().assertTestSuiteResults( 5, 1, 1, 0,
                 0 );
         verifyFailuresOneRetryAllClasses( outputValidator );
 
-        outputValidator = unpack().addGoal( GOAL ).setJUnitVersion( VERSION ).maven().addGoal(
```

Comment:
If you use `debugLogging()` see `build.log` in the folder `target/`. I guess the Surefire prints a message about provider used.